laborator/lab-12/README.md

```diff
@@ -0,0 +1,357 @@
+
+Hex dump of section '.rodata':
+  0x080484d8 03000000 01000200 48656c6c 6f2c2057 ........Hello, W
+  0x080484e8 6f726c64 2100                       orld!.
+```
+
+## Exerciții
+
+Pentru exersarea informațiilor legate de linking, parcurgem mai multe exerciții.
+În cea mai mare parte, aceste exerciții sunt exerciții în care observăm ce se întâmplă în procesul de linking, cele marcate cu sufixul `-tut` sau `-obs`.
+Unele exerciții necesită modificări pentru a repara probleme legate de linking, cele marcate cu sufixul `-fix`.
+Alte exerciții sunt exersarea unor noțiuni (cele marcate cu sufixul `-diy`) sau dezvoltarea / completarea unor fișiere (cele marcate cu sufixul `-dev`).
+Fiecare exercițiu se găsește într-un director indexat; cele mai multe fișiere cod sursă și fișiere `Makefile` sunt deja prezente.
+
+### 01. Linkarea unui singur fișier
+
+Accesăm directorul `01-one-tut/`.
+Vrem să urmărim comenzile de linkare pentru un singur fișier cod sursă C.
+Fișierul sursă este `hello.c`.
+
+În cele trei subdirectoare, se găsesc fișierele de suport pentru următoarele scenarii:
+* `a-dynamic/`: crearea unui fișier executabil dinamic
+* `b-static/`: crearea unui fișier executabil static
+* `c-standalone/`: creare unui fișier executabil standalone, fără biblioteca standard C
+
+În fiecare subdirector folosim comanda `make` pentru a compila fișierul executabil `hello`.
+Folosim comanda `file hello` pentru a urmări daca fișierul este compilat dinamic sau static.
+
+În fișierele `Makefile`, comanda de linkare folosește `gcc`.
+Este comentată o comandă echivalentă care folosește direct `ld`.
+Pentru a urmări folosirea directă a `ld`, putem comenta comanda `gcc` și decomenta comanda `ld`.
+
+În cazul `c-standalone`, pentru că nu folosim biblioteca standard C sau bibliotecă runtime C, trebuie să înlocuim funcționalitățile acestora.
+Funcționalitățile sunt înlocuite în fișierul `start.asm` și `puts.asm`.
+Aceste fișiere implementează, respectiv, funcția / simbolul `_start` și funcția `puts`.
+Funcția / simbolul `_start` este, în mod implicit, entry pointul unui program executabil.
+Funcția `_start` este responsabilă pentru apelul funcției `main` și încheierea programului.
+Pentru că nu există bibliotecă standard, aceste două fișiere sunt scrise în limbaj de asamblare și folosesc apeluri de sistem.
+
+**Bonus**: Adăugați în fișierul `c-standalone/` o comandă care folosește explicit `ld` pentru linkare.
+
+### 02. Linkarea unui singur fișier
+
+Accesați directorul `02-one-diy/`.
+Vrem să compilăm și linkăm fișierele cod sursă din fiecare subdirector, pe modelul exercițiului anterior.
+
+Copiați fișierele `Makefile` și actualizați-le în fiecare subdirector pentru a obține fișierul executabil.
+
+### 03. Linkarea mai multor fișiere
+
+Accesăm directorulA `03-multiple-tut/`.
+Vrem să urmărim comenzile de linkare din fișiere multiple cod sursă C: `main.c`, `add.c`, `sub.c`.
+
+La fel ca în exercițiile de mai sus, sunt trei subdirectoare pentru aceleași scenarii.
```

Comment:
De ce același scenarii? Mi se pare că sunt lucruri diferite.

Reply:
Am modificat puțin textul pe ocw; am menționat că sunt tot trei scenarii, nu aceleași.